pkg/gatherers/clusterconfig/gather_lokistack.go

```diff
@@ -0,0 +1,151 @@
+package clusterconfig
+
+// nolint: dupl
+
+import (
+	"context"
+	"fmt"
+
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/klog/v2"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/client-go/dynamic"
+
+	"github.com/openshift/insights-operator/pkg/record"
+)
+
+const lokiStackResourceLimit = 20
+
+// GatherLokiStack Collects `lokistacks.loki.grafana.com` resources.
+//
+// The gatherer will collect up to 20 resources from `openshift-logging` namespace
+// and it will report errors if it finds a `LokiStack` resource in a different namespace
+// or if there are more than 20 `LokiStacks` in the `openshift-logging` namespace.
+//
+// ### API Reference
+// None
+//
+// ### Sample data
+// - docs/insights-archive-sample/namespaces/openshift-logging/.json
+//
+// ### Location in archive
+// - `namespace/openshift-logging/loki.grafana.com/lokistacks/{name}.json`
+//
+// ### Config ID
+// `clusterconfig/lokistacks
+//
+// ### Released version
+// - 4.18.0
+//
+// ### Backported versions
+// None
+//
+// ### Changes
+// None
+func (g *Gatherer) GatherLokiStack(ctx context.Context) ([]record.Record, []error) {
+	gatherDynamicClient, err := dynamic.NewForConfig(g.gatherKubeConfig)
+	if err != nil {
+		return nil, []error{err}
+	}
+
+	return gatherLokiStack(ctx, gatherDynamicClient)
+}
+
+func gatherLokiStack(ctx context.Context, dynamicClient dynamic.Interface) ([]record.Record, []error) {
+	klog.V(2).Info("Start LokiStack gathering")
+	loggingResourceList, err := dynamicClient.Resource(lokiStackResource).List(ctx, metav1.ListOptions{})
+
+	if errors.IsNotFound(err) {
+		return nil, nil
+	}
+	if err != nil {
+		klog.V(2).Infof("Unable to list %s resource due to: %s", lokiStackResource, err)
+		return nil, []error{err}
+	}
+
+	var records []record.Record
+	var errs []error
+	var otherNamespaceError = false
+	var tooManyResourcesError = false
+
+	for index := range loggingResourceList.Items {
+		item := loggingResourceList.Items[index]
+
+		namespace := item.GetNamespace()
+		if namespace != "openshift-logging" {
+			if !otherNamespaceError {
+				otherNamespaceError = true
+				errs = append(errs, fmt.Errorf("found resource in an unexpected namespace"))
+			}
+
+			continue
+		}
+
+		if len(records) >= lokiStackResourceLimit {
+			if !tooManyResourcesError {
+				errs = append(errs, fmt.Errorf(
+					"found %d resources, limit (%d) reached",
+					len(loggingResourceList.Items), lokiStackResourceLimit),
+				)
+			}
+			continue
```

Comment:
if the maximum number of resources is already triggered, why continuing with the loop instead of breaking it?

Reply:
I did it on purpose, imagine the following scenario:

You have 20 or more LokiStack resources in the `openshift-logging` namespace, and a few ones in a different one. 

If you read first the 20 in the expected namespace and you `break` the loop, you won't be able to detect the other few ones that are in a different namespace

Reply:
Are you ok with this??